src/m365/spo/commands/site/site-remove.ts

```diff
@@ -167,193 +167,170 @@ class SpoSiteRemoveCommand extends SpoCommand {
-        };
-
-        return request.post(requestOptions);
-      })
-      .then((res: string): Promise<string> => {
-        const json: ClientSvcResponse = JSON.parse(res);
-        const response: ClientSvcResponseContents = json[0];
-        if (response.ErrorInfo) {
-          return Promise.reject(response.ErrorInfo.ErrorMessage);
-        }
-        else {
-          const groupId: string = json[json.length - 1].GroupId.replace('/Guid(', '').replace(')/', '');
-          return Promise.resolve(groupId);
-        }
+    const requestOptions: any = {
+      url: `${this.spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+      headers: {
+        'X-RequestDigest': this.context.FormDigestValue
+      },
+      data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="185" ObjectPathId="184" /><Query Id="186" ObjectPathId="184"><Query SelectAllProperties="false"><Properties><Property Name="IsComplete" ScalarProperty="true" /><Property Name="PollingInterval" ScalarProperty="true" /></Properties></Query></Query></Actions><ObjectPaths><Method Id="184" ParentId="175" Name="RemoveDeletedSite"><Parameters><Parameter Type="String">${formatting.escapeXml(url)}</Parameter></Parameters></Method><Constructor Id="175" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+    };
+
+    const response: string = await request.post(requestOptions);
+    const json: ClientSvcResponse = JSON.parse(response);
+    const responseContent: ClientSvcResponseContents = json[0];
+
+    if (responseContent.ErrorInfo) {
+      throw responseContent.ErrorInfo.ErrorMessage;
+    }
+    else {
+      const operation: SpoOperation = json[json.length - 1];
+      const isComplete: boolean = operation.IsComplete;
+
+      if (!wait || isComplete) {
+        return;
+      }
+
+      await new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
+        setTimeout(() => {
+          spo.waitUntilFinished({
+            operationId: JSON.stringify(operation._ObjectIdentity_),
+            siteUrl: this.spoAdminUrl as string,
+            resolve,
+            reject,
+            logger,
+            currentContext: this.context as FormDigestInfo,
+            debug: this.debug,
+            verbose: this.verbose
+          });
+        }, operation.PollingInterval);
       });
+    }
+  }
+
+  private async getSiteGroupId(url: string, logger: Logger): Promise<string> {
+    const _spoAdminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
```

Comment:
Let's set the spoAdminurl to the class property once in the commandAction method.